exercises/collatz-conjecture/.meta/gen.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"log"
+	"text/template"
+
+	"../../../gen"
+)
+
+func main() {
+	t, err := template.New("").Parse(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var j js
+	if err := gen.Gen("collatz-conjecture", &j, t); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type OneCase struct {
+	Description string
+	Number      int
+	Expected    interface{}
+}
+
+// The JSON structure we expect to be able to unmarshal into
+type js struct {
+	Cases []OneCase
+}
+
+func (c OneCase) Valid() bool {
+	valid, _ := determineExpected(c.Expected)
+	return valid
+}
+
+func (c OneCase) Answer() int {
+	_, answer := determineExpected(c.Expected)
+	return answer
+}
+
+// determineExpected examines an .Expected interface{} object and determines
+// whether a test case is valid(bool) and has an answer or expects an error.
+// returning valid and answer.
+func determineExpected(expected interface{}) (bool, int) {
+	ans, ok := expected.(float64)
+	if ok {
+		return ok, int(ans)
+	}
+	return false, 0
```

Comment:
Without if, `return ok, int(ans)` might suffice but as you wish to keep for clarity.

Reply:
I'll keep it.  Thanks @ferhatelmas